test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/io/DatumIonReader.kt

```diff
@@ -240,11 +253,15 @@ class DatumIonReader(
                         }
                         reader.stepOut()
                         val offsetTz = when {
-                            offset == null -> null
-                            offset.isNull -> TimeZone.UnknownTimeZone
-                            else -> TimeZone.UtcOffset.of(offset.int)
+                            offset == null || offset.isNull -> ZoneOffset.UTC
+                            else -> ZoneOffset.ofHoursMinutes(offset.int / 60, offset.int % 60)
                         }
-                        Datum.time(DateTimeValue.time(hour.int, minute.int, second.bigDecimal, offsetTz))
+                        // calculate second/nanos
+                        val ds = second.bigDecimal
+                        val seconds: Int = ds.toInt()
+                        val nanoOfSecond: Int = ds.remainder(BigDecimal.ONE).movePointRight(9).toInt()
```

Comment:
nit: this is used in a few other places. perhaps could be some top-level variable/const?